intel_adsp: Move cavs common files to a subdir

ceolin opened the pull request:
Inside the common directory there were files that are CAVS specific and
are not used by ACE_V1X. Lets create a subdir called cavs inside the
common to put files that are common for only cavs plaftorms.

Note that there are still remaining code that in the common folder that
are using cavs namespace like "cavs_ipc_*" that may need some additional
work.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
intel_adsp: Move cavs common files to a subdir

Inside the common directory there were files that are CAVS specific and
are not used by ACE_V1X. Lets create a subdir called cavs inside the
common to put files that are common for only cavs plaftorms.

Note that there are still remaining code that in the common folder that
are using cavs namespace like "cavs_ipc_*" that may need some additional
work.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

nashif commented:
after this https://github.com/zephyrproject-rtos/zephyr/pull/48142 is merged, those files should go into `soc/xtensa/intel_adsp/cavs/` instead of keeping them in common (they are not common).

ceolin commented:
> after this #48142 is merged, those files should go into `soc/xtensa/intel_adsp/cavs/` instead of keeping them in common (they are not common).

that was my initial idea, than I thought would be confuse to have `intel_adsp/cavsX` and `intel_adsp/cavs`. As the code is "common" to cavs platforms I just created a folder inside the common. Anyway, lets the other pr gets in and change it.

nashif commented:
> that was my initial idea, than I thought would be confuse to have `intel_adsp/cavsX`

what is `intel_adsp/cavsX` and why would that be needed?

If the code is common to cavs platforms, it belongs into `soc/xtensa/intel_adsp/cavs/`, just like we have code common to ace under `soc/xtensa/intel_adsp/ace_v1x/`. Note that I still ldid not change the name of `ace_v1x` as a series, something to do moving forward, but this should ideally end up as `soc/xtensa/intel_adsp/ace/` ....

Commit:
intel_adsp: Rename multiprocessing cavs file

For intel adsp platforms we have a common multiprocessing.c file and
then another multiprocessing.c for ace and multiprocessing_cavs.c for
cavs. Rename the cavs specific implementation to follow the convention.
There is not need to "cavs" suffix since the file is already inside its
specific directory.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>
